src/utils/hsdis/binutils/hsdis-binutils.c

```diff
@@ -244,9 +246,9 @@ static const char* format_insn_close(const char* close,
 
   strcpy(buf, close);
   char* p = buf;
-  if (type)    sprintf(p += strlen(p), " type='%s'", type);
-  if (dsize)   sprintf(p += strlen(p), " dsize='%d'", dsize);
-  if (delays)  sprintf(p += strlen(p), " delay='%d'", delays);
+  if (type)    snprintf(p += strlen(p), remaining_buflen(buf, bufsize, p), " type='%s'", type);
+  if (dsize)   snprintf(p += strlen(p), remaining_buflen(buf, bufsize, p), " dsize='%d'", dsize);
+  if (delays)  snprintf(p += strlen(p), remaining_buflen(buf, bufsize, p), " delay='%d'", delays);
```

Comment:
What is the value for p that is used in the call to remaining_buflen?
It is being assumed to be the value after the assignment by the first
argument. However, according to the standard, it is unspecified, and
the whole snprintf call invokes UB. This is because there aren't any
sequence points between the update of p in the first argument and that
reference. (C++17 changes this, but we aren't using C++17 yet.)

Reply:
For the 1st line (line 249), it is fine to use bufsize directly. The p in remaining_buflen is used to calculate the remaining length of the target buffer.  To know the remaining buffer length, I need to know the pointer to the beginning buffer (the 'buf' parameter), the pointer to the beginning of the unused buffer (the 'p' parameter), and the total size of the buffer (the 'bufsize' parameter).  The initial value of p is assigned to buf(p = buf), and then move forward by the used size ( p += strlen(p)).

It's a good point to me about the update sequence of p, and I will make an update in a new PR.